File: ColtSteele/power.py
"""
Description: compute power with a base and expo
Input: base , power
Ouput: base raised to that power
"""

def power1(base,expo):
    if expo == 1:
        return base
        
    return base * power1(base,expo-1)

if __name__ == "__main__":
    print(power1(2,4))

File: 2020_05_16_Start/Java/src/com/JavaAlgos/ColtSteele/Recursion/Power.java
package com.JavaAlgos.ColtSteele.Recursion;

public class Power {
    public static void main(String[] args){
        System.out.println(findPower(2,0));
        System.out.println(findPower(2,2));
        System.out.println(findPower(2,4));
        System.out.println(findPower(2,100));
    }

    public static int findPower(int base, int expo){
        if(expo == 0) return 1;
        return base*findPower(base,expo-1);
    }
}
